### ISSUE TITLE
Autoupload for nextcloud dev (27/07) choose filename

### ISSUE THREAD

**Issue by ruvashev** [nextcloud/android#1239]
### Actual behaviour
Autoupload uploads images/video files to nextcloud server keeping the same name.
So if delete after upload option is selected android will reset DSC_XXXX file seq from the most recent.
Example files in DCIM folder. DSC_0001, DSC_0002.
I take a photo, filename is DSC_0003.
Nextcloud detects it, uploads, deletes.
Next time i take the photo again and file name again is DSC_0003 => clash at autoupload.

### Expected behaviour
It would be nice to name files at upload like it is done in iOS version or in previous version like 1.4.0-something
Photo date time taken DSC seq number.
Or give option to the user to choose how to name it.
 
### Steps to reproduce
1. setup autoupload the way it delete photo after it is taken
2. take photo, wait autoupload to upload it
3. take another photo, it will not get uploaded.

### Environment data
Android version: 4.4.4

Device model: xperia Z2/D6503

Stock or customized system: rooted

Nextcloud app version: dev latest 07/27

Nextcloud server version: 11.0.0

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by mario** [comment]
With the new auto upload implementation, it would overwrite the existing file on the server.

**Comment by ruvashev** [comment]
correct, however it would be beneficial to implement complex file name on the server like "Photo Date Time taken seq number" that way regardless of current file seq number we will have both files on the server. Besides:
- that file naming was already implemented in earlier versions of nextcloud app
- it is implemented in iOS
- it facilitates sorting of files by name.
Can we please implement it?

**Comment by tobiasKaminsky** [comment]
We never had a file naming support and I am unsure if changing the file name should be part of this app.
For me a sync app should sync, but not modify any files.
Renaming the images to a time based name should be done by the camera app.

**Comment by ruvashev** [comment]
Thanks Mario and Tobias. I have an example: Photo-2017-01-10-00-07-24_4078.JPG. Since it was never in nextcloud app then i might have mixed it up with owncloud app that i used initially till I switched to nextcloud. Sorry for confusion.

Thanks for considering as enhancements.

**Comment by teras** [comment]
It might indeed be a problem with the camera app, but unfortunately this is the way the default camera app on android is working. Changing the camera app sometimes isn't even possible (not technically, mostly practically).
Thus I believe special care should be taken for auto upload folders and files. Maybe auto-upload should have a timestamp by default?

**Comment by caco3** [comment]
Is there any progress on this issue?

I have exactly the same problem. I would have lost a lot of photos if the files version function would not have worked!

My usecase:
I use a smartphone where the stock camera names the images with the format DSC_0001.JPG. Since the device is unrooted, I can sadly not improve this ugly behaviour.
To save memory, I let the nextcloud client move the images to my SD card after uploading them.
How ever this makes the camera app to start at number 1 again and again.
The nextcloud app blindly overwrites the files on the server then...

NB: 

- The camera could directly write to the SD card, how ever this is not reliable and lead to issues sometimes (SD card not mounted).
 - Keeping the photos in the folder instead of moving them to the nextcloud client folder makes it hard to see which photos got already uploaded. I had used it this way but had experienced photos not uploaded and always a full internal memory.
- There is a tool "DSC Auto rename" (https://play.google.com/store/apps/details?id=ro.ciubex.dscautorename&hl=de) which can rename the files automatically, how ever it interferes with the auto upload functionality.

Suggested solutions:

- The nextcloud app could have an option to rename files if they have such filename format.
- The nextcloud app could provide a delay so DSC auto renamer could do its work first.
- The nextcloud app could provide an option to rename files if they exist on the server

**Comment by noboevbo** [comment]
I had roughly implemented the function in a fork (UI adjustments are still necessary, currently only a text field), but I don't think the function is desired this way, because there was no more feedback.

**Comment by caco3** [comment]
I went now for a server side solution using the workflow app and a shell script. When ever a file gets created, the script gets triggered and renames the file (if needed). At the same time it also compresses it.

**Comment by durianwool** [comment]
Hi, I like this feature now that it has been implemented. However, I can't find documentation anywhere on how to retain the original file name. That is on iOS, I've had it set to "YYYY-MM-DD_HH-mm_", and the resulting name is "2021-05-23_22-59_0123". The original in the phone was "IMG_0123". How can I specify it to retain the original name even after the mask? That is, "2021-05-23_22-59_**IMG_**0123

**Comment by AndyScherzinger** [comment]
@durianwool Android doesn't support this feature and from your text I guess you are looking for the iOS tracker: https://github.com/nextcloud/ios/issues

**Comment by coxde** [comment]
Is there any update? It's great to have this feature in iOS, but not in Android, unfortunately.

**Comment by mawi42** [comment]
I am also waiting on that feature, is there a plan to implement it or not?
